eml_parser/regexes.py

```diff
@@ -37,7 +37,39 @@
 #   - do not use a fixed list of TLDs but rather \w
 #   - only check for URLs with scheme
 #   - modify the end marker to allow any acceptable char according to the RFC3986
-url_regex_simple = re.compile(r'''(?i)\b(?:https?|ftps?):(?:/{1,3}|[a-z0-9%])(?:[^\s()<>{}\[\]]+|\([^\s()]*?\([^\s()]+\)[^\s()]*?\)|\([^\s]+?\))+(?:[\w\-._~%!$&'()*+,;=:/?#\[\]@]+)''')
+
+try:
```

Comment:
Could you please add an explanation as to why you include two slightly different re.compile blocks. Which one works/doesn't for what lib.

Reply:
I have changed the block from a try/except to if/else based on the re.__name__. `re2` uses `\x{0000}` instead of `\u0000` and '\U00000000`.  This is a simple enough change that one string could be compiled after a re.sub() modifies the string, but there are other implementation differences between re, re2, and regex that may require entirely separate expressions.